feat(detectExecuteScan): Add `useDetect9` for latest BlackDuck script

bugwelle opened the pull request:
`detect8.sh` is no longer the most recent version of the BlackDuck runner. `detect9.sh` is now available and fixes some important bugs such as incorrect dependency reporting for `package-lock.json`'s in version 2 and 3.

This commit adds `detect9.sh` support by adding a new `detectExecuteScan` option `useDetect9`, similar to the old option `useDetect8`.

See release notes of Synopsis Detect:
 - https://sig-product-docs.synopsys.com/bundle/integrations-detect/page/currentreleasenotes.html

# Changes

- [ ] Tests
- [ ] Documentation

CCFenner commented on resources/metadata/detectExecuteScan.yaml:
```diff
@@ -324,6 +324,17 @@ spec:
           - STEPS
         default: true
         deprecated: true
```
Why is this deprecated?

bugwelle replied:
I don't know. I'd remove the flag if it were up to me. 😄 
Maybe someone from the BlackDuck team can answer. I see that none is assigned, though. :)

dimaste replied:
it's just placeholder at the moment, not used value

bugwelle commented:
@CCFenner Is there anything else I can do? Or do I need a review by someone else? If so, could you assign/ping them? :)
I don't have permission to merge this, regardless of your approval and passed CI tests.

Regards,
Andre

dimaste commented:
This is the default value, but we don't use it actually. It works in any case where no other detect version parameter is provided. 

> @dimaste Sure! Done. Btw: Isn't detect8 still the default? Or was the `default: false` change only for documentation? 😄

dimaste commented:
@bugwelle we discussed your PR with product management. We can merge it after the server upgrade which we should complete by the end of the year. The reason is that the current server version does not officially support it. 

I'll merge it once the server is upgraded. For now, let's keep the PR.

bugwelle commented:
> The reason is that the current server version does not officially support it.

Oh! 😄 
We're already using detect9.sh for some projects and didn't notice anything off.

Thanks for reviewing it. :)

dimaste commented:
Sure I see your concern, you can try to pass this as a workaround in Piper config of detect step for your scenario 

customEnvironmentVariables:
      - 'DETECT_LATEST_RELEASE_VERSION=9.1.0'

bugwelle commented:
@dimaste  I've updated the PR. It seems that useDetect7/8 was removed during the last 3 months. I've added `useDetect9` again.

bugwelle commented:
@dimaste  I've rebased the PR. It seems that useDetect7/8 was removed during the last 5 months. I've added `useDetect9` again.

@SAP/jenkins-library-team can you remove the stale label?

dimaste commented:
Hi @bugwelle , this functionality has been merged separately so you can use it :)
